test/cypress/integration/widgets/secretProviders_spec.ts

```diff
@@ -0,0 +1,28 @@
+describe('Secret Providers widget', () => {
+    const widgetId = 'secretProviders';
+    const widgetConfiguration = {
+        pollingTime: 3,
+        pageSize: 0 // NOTE: Setting page size to 0 to list all secret providers and be able to find the one created in test
+    };
+
+    before(() => {
+        cy.activate('valid_trial_license').usePageMock(widgetId, widgetConfiguration).mockLogin();
+        cy.fixture('secret_providers/secret_providers').then(secretProviders => {
+            cy.interceptSp('GET', '/secrets-providers*', {
+                items: secretProviders,
+                metadata: {
+                    pagination: {
+                        total: 1,
+                        size: 1,
+                        offset: 0
+                    }
+                }
+            });
+        });
```

Comment:
Two things here:
1. Changing the order of lines (L9 & L10) would make this test more reliable (as we will already be prepared with intercept before login)
2. `total` and `size` seems not accurate as we actually have 2 secret providers in the fixture 🙂 

WDYT about something like that:
```suggestion
        cy.fixture('secret_providers/secret_providers').then(secretProviders => {
            const numberOfSecretProviders = secretProviders.length;
            cy.interceptSp('GET', '/secrets-providers*', {
                items: secretProviders,
                metadata: {
                    pagination: {
                        total: numberOfSecretProviders,
                        size: numberOfSecretProviders,
                        offset: 0
                    }
                }
            });
        });
        cy.activate('valid_trial_license').usePageMock(widgetId, widgetConfiguration).mockLogin();
```
?

Alterantively you can move `metadata` object to the fixture.